src/wallet/scriptpubkeyman.cpp

```diff
@@ -1770,7 +1770,7 @@ bool DescriptorScriptPubKeyMan::GetReservedDestination(const OutputType type, bo
     } else {
         error = util::ErrorString(op_dest);
     }
-    return bool(op_dest);
+    return op_dest.has_value();
```

Comment:
Thanks for picking up the suggestion https://github.com/bitcoin/bitcoin/pull/25721#discussion_r938825139 here.